### PR TITLE
Add support for extension GL_ARB_shader_storage_buffer_object.

### PR DESCRIPTION
**Pull request by pmistryNV**
Enable below features for GL Core version 420:
* layout qualifier "std430"
* storage qualifier "buffer"
* atomic memory functions

### CODE REVIEW

**Comment by pmistryNV** [comment]
@johnkslang kindly review the changes.

File: glslang/MachineIndependent/Scan.cpp [hunk 1]
```diff
@@ -917,7 +917,7 @@ int TScanContext::tokenizeIdentifier()
     case BUFFER:
         afterBuffer = true;
         if ((parseContext.isEsProfile() && parseContext.version < 310) ||
-            (!parseContext.isEsProfile() && parseContext.version < 430))
+            (!parseContext.isEsProfile() && parseContext.version < 420))
```
Comment (johnkslang) [inline glslang/MachineIndependent/Scan.cpp:920]: Is this incorrectly classifying something as a keyword even if the extension is not enabled?

Comment (pmistryNV) [reply glslang/MachineIndependent/Scan.cpp:920]: This extensions makes the BUFFER token legal for version 420. However at this point we cannot determine if the extension is enabled. This is determined later in TParseContext::setLayoutQualifier.

Comment (johnkslang) [reply glslang/MachineIndependent/Scan.cpp:920]: You can test that the extension is enabled, right here, before deciding it is a keyword.  You can see some other examples of this in that neighborhood.  Look for `parseContext.extensionTurnedOn(...`.  Best to do it after the simple testing, as it is a bit expensive.

Comment (pmistryNV) [reply glslang/MachineIndependent/Scan.cpp:920]: Sure I can add it, but this would also mean compiler will not proceed further and flag errors beyond it.

Comment (johnkslang) [reply glslang/MachineIndependent/Scan.cpp]: I don't understand.  Should buffer be a keyword or not depending on whether the extension is enabled?  Why would that give the wrong result?

Comment (pmistryNV) [reply glslang/MachineIndependent/Scan.cpp]: Yeah I understood what you meant. I have updated the patch.

**Pull request merged**